src/cmd-buildextend-gcp

```diff
@@ -7,94 +7,160 @@ extending the meta.json with GCP image name.
-parser.add_argument("--gce", help="Use GCE as the platform ID instead of GCP",
-                    action="store_true",
-                    default=bool(os.environ.get("GCP_GCE_PLATFORM_ID", False)))
-parser.add_argument("--force", help="Replace existing images and upload",
-                    action="store_true",
-                    default=bool(os.environ.get("GCP_FORCE", False)))
-parser.add_argument("--json-key", help="GCP Service Account JSON Auth",
-                    default=os.environ.get("GCP_JSON_AUTH"))
-parser.add_argument("--name-suffix", help="Append suffix to name",
-                    required=False)
-parser.add_argument("--project", help="GCP Project name",
-                    default=os.environ.get("GCP_PROJECT_NAME"))
-args = parser.parse_args()
-
-# Argument checks for environment strings that are required
-arg_exp_str = "parameter '--{}' or envVar '{}' must be defined"
-if args.bucket is None:
-    raise Exception(arg_exp_str.format("bucket", "GCP_BUCKET"))
-if args.json_key is None:
-    raise Exception(arg_exp_str.format("json-key", "GCP_JSON_AUTH"))
-if args.project is None:
-    raise Exception(arg_exp_str.format("project", "GCP_PROJECT"))
-
-# Identify the builds
-builddir = os.path.join('builds', args.build)
-buildmeta_path = os.path.join(builddir, 'meta.json')
-with open(buildmeta_path) as f:
-    buildmeta = json.load(f)
-
-if 'gcp' in buildmeta['images'] and not args.force:
-  print("GCP image already built; use --force to rebuild")
-  sys.exit(0)
-
-# Name the base build and tarball name
-base_name = buildmeta['name']
-if args.name_suffix:
-    gcp_name_version = f'{base_name}-{args.name_suffix}-{args.build}'
-else:
-    gcp_name_version = f'{base_name}-{args.build}'
-gcp_tarball_name = f'{gcp_name_version}-gcp.tar.gz'
-
-# Setup the tempdir
-tmpdir = 'tmp/buildpost-gcp'
-if os.path.isdir(tmpdir):
-    shutil.rmtree(tmpdir)
-os.mkdir(tmpdir)
-tmpdir = os.path.abspath(tmpdir)
-
-def generate_gcp_tar():
-    """ Set the Ignition ID to GCP/GCE and create a tarball for upload """
-    plat_id = 'gcp'
+        sha256sum_file)
+
+from cosalib.build import _Build
+
+# Temporary directory used for this command
+TMPDIR = os.path.abspath('tmp/buildpost-gcp')
+# Identifier for gce platform
+PLATFORM_GCE = 'gce'
+# Identifier for gcp platform
+PLATFORM_GCP = 'gcp'
+
+
+class Build(_Build):
+    """
+    GCP implementation of Build.
+    """
+
+    def _create_tmp_dir(self):
+        """
+        Ensures the temp directory is created and clean.
+        """
+        # Setup the tempdir
+        if os.path.isdir(TMPDIR):
+            shutil.rmtree(TMPDIR)
+        os.mkdir(TMPDIR)
+
+    def _build_artifacts(self, *args, **kwargs):
+        """
+        Implements the building of artifacts. Walk the build root and
+        prepare a list of files in it.
+
+        :param args: All non-keyword arguments
+        :type args: list
+        :param kwargs: All keyword arguments
+        :type kwargs: dict
```

Comment:
This is totally my personal feeling but...I feel like part of the value of Python is that it's not too verbose, one can write things quickly.  These "types in doc comments" make it more of a mixed bag.  If I was writing a major Python library I'd probably do it, but in this case it should be totally obvious to anyone who knows Python at all that `**kwargs` is a dict of args.

Not saying we should drop this at all, but hope no one objects if I don't add similar doc comments myself :wink:  (If we are going to go for verbosity I'd vote for a static language instead)

Reply:
Absolutely fair. I tend to do this more so because of sphinx documentation, but it's not required at all IMHO :-)